src/utils.js

```diff
@@ -91,13 +93,24 @@ export function setupYupExtensions() {
     });
   });
 
-  Yup.addMethod(Yup.string, 'compare', function compare(message) {
+  Yup.addMethod(Yup.string, 'compare', function compare(message, type) {
     return this.test('isGreater', message, function isGreater() {
-      if (!this.parent || !this.parent.startTime || !this.parent.endTime) {
+      if (!this.parent
```

Comment:
I generally don't like comments explaining code, but I think if we have one here can be very helpful.